### PR TITLE
Currency API Base

### PR DESCRIPTION
**Pull request by petrspelos**
## Changes
This is an implementation of the basic features needed. Everything optional is still available for others to implement.

* New `CZK` status command
b

* New `Convert XX.XX YYY to ZZZ` where either YYY or ZZZ must be `CZK`. (for now)
a

bitmoji

### CODE REVIEW

File: src/Miunie.Core/Services/CurrencyService.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+namespace Miunie.Core
+{
+    public class CurrencyService
+    {
+        private const string DailyConversionApiUrl = @"https://www.cnb.cz/cs/financni_trhy/devizovy_trh/kurzy_devizoveho_trhu/denni_kurz.xml";
+        private const string DailyConversionApiUrlDateParam = @"?date=";
+        private const string RowElement = "radek";
+        private const string AmountAttribute = "mnozstvi";
+        private const string CodeAttribute = "kod";
+        private const string CzechCrownsAttribute = "kurz";
```
Comment (svr333) [inline src/Miunie.Core/Services/CurrencyService.cs]: At first I was really wondering what these names refer to, then I realized you're using a Czech Republic site.
All good 😅

File: src/Miunie.Discord/CommandModules/CurrencyCommand.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        private readonly CurrencyService _currencyService;
+
+        public CurrencyCommand(EntityConvertor convertor, CurrencyService currencyService)
+        {
+            _entityConvertor = convertor;
+            _currencyService = currencyService;
+        }
+
+        [Command("CZK")]
+        public async Task AddReputationAsync(CommandContext ctx)
```
Comment (svr333) [inline src/Miunie.Discord/CommandModules/CurrencyCommand.cs]: I'm pretty sure the method name isn't correct.

File: src/Miunie.Discord/CommandModules/CurrencyCommand.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            var channel = _entityConvertor.ConvertChannel(ctx.Channel);
+            await _currencyService.ShowCzkStatus(channel);
+        }
+
+        [Command("CZK")]
+        public async Task AddReputationAsync(CommandContext ctx, string dateTime)
```
Comment (svr333) [inline src/Miunie.Discord/CommandModules/CurrencyCommand.cs]: Same here.

**Comment by petrspelos** [comment]
Thanks, @svr333 that's what happens when you copy paste. xD
bitmoji

**Pull request merged**